pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -733,11 +733,28 @@ func getNodeOrPVCName(d *appsv1.Deployment) (string, error) {
 	if v, ok := d.Labels[OSDOverPVCLabelKey]; ok {
 		return v, nil // OSD is on PVC
 	}
+	for k, v := range d.Spec.Template.Spec.NodeSelector {
+		if k == k8sutil.LabelHostname() {
+			return v, nil
+		}
+	}
+	// try to fallback on default hostname label.
+	// needed to perform upgrde to custom hostname node label:
+	// question: how to handle custom label update? We need previous label name here.
+	// in case of non-portable osd NodeSelector always have one entry and it is easy to get previous label.
+	// but how to handle portable OSD?
+	// https://github.com/rook/rook/blob/a9af67d68532b5686c588f0f7c650e24df2432ac/pkg/operator/ceph/cluster/osd/spec.go#L754-L761
 	for k, v := range d.Spec.Template.Spec.NodeSelector {
```

Comment:
I don't believe we need these lines 747-756. The loop in 747-751 seems covered above in 736-740, and not sure I understand the need for the other loop.

Reply:
Correct. At first, i was not sure because of portable OSD but now i see that there will be always one label in NodeSelector.